src/v2/Apps/Shows/Routes/ShowsCity.tsx

```diff
@@ -65,25 +65,27 @@ export const ShowsCity: React.FC<ShowsCityProps> = ({
     offset: 20,
   })
 
-  const handleClick = (cursor: string) => {
+  const handleClick = (cursor: string, page: number) => {
     scrollTo()
 
     setLoading(true)
 
     relay.refetch(
-      { slug: city.slug, first: 18, after: cursor },
+      { slug: city.slug, first: 18, after: cursor, page: null },
       null,
       error => {
         if (error) {
           console.error(error)
         }
 
         setLoading(false)
+
+        window.history.replaceState({}, "", `?page=${page}`)
```

Comment:
Should we be using `router.replace`?

Reply:
That triggers a full page render (and disrupts the scroll). This is apparently how we handle things in the artwork filter, for instance.